examples/react/react-app/src/dojo/dojo_starter.ts

```diff
@@ -0,0 +1,154 @@
+// This is an example of a generated TS file that exposes the types and methods for the Dojo starter contract.
+// Currently hard-coded to serve as a reference.
+
+// Utility type to extract common properties from a list of objects
+// This is used for the parameters of the `findEntities` method
+type CommonProps<T extends object[]> = T extends [infer First, ...infer Rest]
+    ? Partial<
+          {
+              [K in keyof First &
+                  keyof Rest[number]]: First[K] extends Rest[number][K]
+                  ? { [P in K]: First[K] }
+                  : never;
+          }[keyof First & keyof Rest[number]]
+      >
+    : never;
+
+//
+//
+// The base class for the Dojo codebase
+//
+//
+
+interface InitialParams {
+    rpcUrl: string;
+    toriiUrl: string;
+}
+
+// Auto-generated name from the Scarb.toml
+export class Dojo_Starter {
+    rpcUrl: string;
+    toriiUrl: string;
+    moves: MovesQueries;
+    position: PositionQueries;
+    actions: ActionsCalls;
+
+    constructor(params: InitialParams) {
+        this.rpcUrl = params.rpcUrl;
+        this.toriiUrl = params.toriiUrl;
+        this.moves = new MovesQueries();
+        this.position = new PositionQueries();
+        this.actions = new ActionsCalls();
+    }
+
+    findEntities<T extends object[]>(query: CommonProps<T>): T[number][] {
+        // Construct the return type dynamically based on the generic type parameter T
+        const test: T[number][] = [];
+
+        return test;
+    }
+}
+
+//
+//
+// Entity types generated from the models
+//
+//
+
+interface BaseEntity {
```

Comment:
Should this be BaseModel? And also a few other instances below.

Reply:
Well, that's what I had at first. But when fetching a single entity, it felt weird that the return type was of `Model`, when in essence you were fetching an entity. WDYT?

![CleanShot 2024-03-11 at 10 14 24](https://github.com/dojoengine/dojo.js/assets/11869830/6db43089-cd16-4c4d-9c46-8a8726e4ad8b)

Reply:
Seems like this was because of my confusion with the whole data model. Updated now!